R/mergeData.R

```diff
@@ -32,20 +32,34 @@
 
 mergeData <- function(feat_eng_df, clinical_data, match = "ScrID", id_mapping = NULL) {
 
-  # note: still need to generalize this to more colnames.
   if (!is.null(id_mapping)) {
+
+  # converting literally everything into a factor
+  id_index <- grep(match, colnames(id_mapping))
+  id_mapping[, id_index] <- as.factor(id_mapping[, id_index])
+  feat_eng_df$pid <- as.factor(feat_eng_df$pid)
+  clinical_data$PatID <- as.factor(clinical_data$PatID)
```

Comment:
How do you know that the patient ID column in `clinical_data` is `PatID`? I assume `pid` is the default output of the feature engineering df, correct?